lib/Scheduling/TestPasses.cpp

```diff
@@ -361,6 +364,52 @@ void TestSimplexSchedulerPass::runOnFunction() {
   llvm_unreachable("Unsupported scheduling problem");
 }
 
+//===----------------------------------------------------------------------===//
+// SchedulableOpInterface
+//===----------------------------------------------------------------------===//
+
+namespace {
+struct TestSchedulableOpInterfacePass
+    : public PassWrapper<TestSchedulableOpInterfacePass, FunctionPass> {
+  void runOnFunction() override;
+  StringRef getArgument() const override {
+    return "test-schedulable-op-interface";
+  }
+  StringRef getDescription() const override {
+    return "Test the schedulable op interface.";
+  }
+};
+
+struct CombSchedulable
+    : public SchedulableOpInterface::FallbackModel<CombSchedulable> {
+  unsigned getLatency(Operation *op) const { return 0; }
+};
+
+struct ThreeCycleSchedulable
+    : public SchedulableOpInterface::FallbackModel<ThreeCycleSchedulable> {
+  unsigned getLatency(Operation *op) const { return 3; }
+};
+
+struct TestComponentLibrary {
+  TestComponentLibrary(MLIRContext &context) {
+    AddIOp::attachInterface<CombSchedulable>(context);
```

Comment:
Very nice! I wasn't aware of this option - removes all concerns i had about _"How do we implement interfaces for operators which aren't related to what we want to do, if looking at the pure MLIR level?"_.